pkg/logs/processor/processor.go

```diff
@@ -53,16 +54,23 @@ func (p *Processor) Stop() {
 }
 
 // Flush processes synchronously the messages that this processor has to process.
-func (p *Processor) Flush() {
+func (p *Processor) Flush(ctx context.Context) {
 	p.mu.Lock()
-	for {
-		if len(p.inputChan) == 0 {
-			break
+	defer p.mu.Unlock()
+	running := true
+	for running {
+		select {
+		case <-ctx.Done():
+			return
+		default:
```

Comment:
This we're using a `default we could simply do:

```go
if ctx.Err() != nil {
    return
}
```

And remove the switch. I thinks it's clearer and make it explicit that the `default` is blocking and can't be canceled. WDYT ?

Reply:
This `select` + `case <-ctx.Done()` is a well known pattern while using `context.Context` for cancellation (even the one documented as far as I recall but not entirely sure anymore).